cpp/src/c_api/legacy_k_truss.cpp

```diff
@@ -81,27 +81,13 @@ struct k_truss_functor : public cugraph::c_api::abstract_functor {
       auto number_map = reinterpret_cast<rmm::device_uvector<vertex_t>*>(graph_->number_map_);
 
       auto graph_view = graph->view();
-      rmm::device_uvector<vertex_t> src(0, handle_.get_stream());
-      rmm::device_uvector<vertex_t> dst(0, handle_.get_stream());
-      std::optional<rmm::device_uvector<weight_t>> wgt{std::nullopt};
 
-      std::tie(src, dst, wgt, std::ignore) = cugraph::decompress_to_edgelist(
+      auto [result_src, result_dst] = cugraph::k_truss<vertex_t, edge_t, multi_gpu>(
```

Comment:
You are replacing the call to legacy ktruss with the call to primitive-based ktruss.  That's great in that it saves python work (although we should still plan MG python work for 24.06).  There's already issues for that work, we can update those issues.

However, if we do that, we should rename this file to `k_truss.cpp` since it will no longer be the legacy implementation.

Reply:
Now we should start addressing all the issues we should address before merging this PR.